src/ui/Templates/TemplateList.ts

```diff
@@ -23,8 +32,18 @@ export class TemplateList extends Template {
 
   instantiateToElement(object: IQueryResult, instantiateOptions: IInstantiateTemplateOptions = {}): HTMLElement {
     let merged = new DefaultInstantiateTemplateOptions().merge(instantiateOptions);
-    for (var i = 0; i < this.templates.length; i++) {
-      var element = this.templates[i].instantiateToElement(object, merged);
+
+    if (merged.role != null) {
+      const roledTemplate = _.find(this.templates, t => t.role === merged.role);
```

Comment:
Once again, I think we are over complicating this class, with it now needing to support "normal" templates, and templates needed for a very narrow/precise role for header/footer.

Those template don't need all the complexity of a "normal" template with condition evaluation (they are always loaded).

I have the gut feeling that we could have something much more simple if we had dedicated class/modules for table layouts in general.